raami/queries.js

```diff
@@ -0,0 +1,269 @@
+//     .then(function (result) {
+//       /* jshint ignore:start */
+//       res.status(200)
+//         .json({
+//           status: 'success',
+//           message: `Removed ${result.rowCount} puppy`
+//         });
+//       /* jshint ignore:end */
+//     })
+//     .catch(function (err) {
+//       return next(err);
+//     });
+// }
+
+/**** WORKS ***/
+
+function getWorks(req, res, next) {
+  var _id = (req.params.id);
+  req.app.locals.db.any(`
+    SELECT id, artist_id, title, width, height, depth, gallery, orientation, technique,
+           filename, filedata, year, price
+      FROM Works
+     WHERE artist_id = $1`, _id
+  )
+    .then(function (data) {
+      res.status(200)
+        .json({
+          works: data,
+        });
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+function getWork(req, res, next) {
+  var _id = (req.params.id);
+  req.app.locals.db.one(`
+    SELECT artist_id, title, width, height, depth, gallery, orientation, technique,
+           filename, filedata, year, price
+      FROM Works
+     WHERE id = $1`, _id
+  )
+    .then(function (data) {
+      res.status(200)
+        .json(data);
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+
+function createWork(req, res, next) {
+  req.app.locals.db.one(`
+    INSERT INTO Works
+                (
+                  artist_id, title, width, height, depth, gallery, orientation,
+                  technique, filename, filedata, year, price
+                )
+         VALUES (
+                  $(artist_id), $(title), $(width), $(height), $(depth), $(gallery),
+                  $(orientation), $(technique), $(filename), $(filedata),
+                  $(year), $(price)
+                )
+        RETURNING id`, req.body
+  )
+    .then(function (data) {
+      res.status(200)
+        .json({
+          status: 'success',
+          insrted: data.id
+        });
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+
+function updateWork(req, res, next) {
+  var _id = (req.params.id)	
+  req.app.locals.db.none(`
+    UPDATE Works
+       SET artist_id=$1, title=$2, width=$3, height=$4, depth=$5, gallery=$6, filename=$7,
+           filedata=$8, price=$9, year=$10, orientation=$11, technique=$12
+     WHERE id=$13`, [
+      req.body.artist_id,
+      req.body.title,
+      req.body.width,
+      req.body.height,
+      req.body.depth,
+      req.body.gallery,
+      req.body.filename,
+      req.body.filedata,
+      req.body.price,
+      req.body.year,
+      req.body.orientation,
+      req.body.technique,
+      _id
+    ]
+  )
+    .then(function () {
+      res.status(200)
+        .json({
+          status: 'success',
+          updated: req.body
+        });
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+function removeWork(req, res, next) {
+  var _id = (req.params.id);
+  req.app.locals.db.result('DELETE FROM Works WHERE id = $1', _id)
```

Comment:
Should require `raami_admin` rights, or check that the email address of `person_id` matches `req.session.user.email`.